test/cmdlineTests/standard_optimizer_yulDetails_optimiserSteps_with_empty_cleanup_sequence/input.json

```diff
@@ -0,0 +1,16 @@
+{
+    "language": "Solidity",
+    "sources": {
+        "A": {"urls": ["standard_optimizer_yulDetails_optimiserSteps_with_cleanup_sequence/in.sol"]}
+    },
+    "settings": {
+        "optimizer": {
+            "details": {
+                "yul": true,
+                "yulDetails": {
+                    "optimizerSteps": "dhfoDgvulfnTUtnIf\n[    xarrscLM\n]\njmuljuljul VcTOcul jmul:"
```

Comment:
One thing that would be really nice to have covered by tests is the resulting sequence here. We want to be sure that this clears the cleanup sequence and does not use the default. Currently we won't spot that because these tests do not have any outputs selected that would depend on the sequence.

One way to do that would be to request the `metadata` output since that includes the parsed and reserialized sequence. The problem is that metadata contains the compiler version string, which changes based on build configuration so the test will be passing on some configurations in CI but not others. We do have some limited [metadata stripping in `cmdlineTests.sh`](https://github.com/ethereum/solidity/blob/v0.8.16/test/cmdlineTests.sh#L189-L241) exactly for that reason but that only applies to bytecode and strips the whole metadata anyway. Would be nice to extend that so that it can remove the version field - but that would be better off as a separate PR.

A better option would be to add some boost tests in [`test/solc/CommandLineParser.cpp`](https://github.com/ethereum/solidity/blob/develop/test/solc/CommandLineParser.cpp). There you can pass in parameters and inspect the resulting settings object.

Reply:
Looking at the non-JSON tests I see it does get tested to some extent because there we have the optimized IR output. In that case we should probably request the optimized IR output here too. I'd make the input snippets identical to CLI ones as well.

Still, I think that having this tested in `test/solc/CommandLineParser.cpp` would be a good idea. It's much easier to notice that the sequence got mangled than that the IR output went through unexpected optimizer steps.